components/storage/StorageTable.tsx

```diff
@@ -79,25 +79,52 @@ const Table: React.FC<TableProps> = ({ services, selectedAnswers, questions }) =
   const columnHelper = createColumnHelper<TableRow>();
   const tableContainerRef = useRef<HTMLDivElement>(null);
 
-const tableData: TableRow[] = useMemo(() => {
-  const uniqueFeatureNames = new Set<string>();
-  services.forEach(service => {
+  // --- FEATURE SORTING LOGIC ---
+  const sortedFeatureNames = useMemo(() => {
```

Comment:
I think i'd like to walk through this function with you

Why do you need the `useMemo`?

Reply:
well, i think given that the table takes so long to load it maybe isn't really helping at all. but i thought that it is supposed to help with performance. so at each time the page re-renders, `useMemo()` only runs that code whenever the dependencies have changed. So in this case, all of this sorting code only runs if `sortedFeatureNames` has changed